ESLint warnings in file that doesn't exist

## Description

I'm getting ESLint warnings in a file that doesn't seem to exist when adding and configuring ESLint using https://www.gatsbyjs.org/packages/gatsby-plugin-eslint/ and https://www.npmjs.com/package/eslint-plugin-react-app.

```sh
warn ESLintError:
/Users/mrseanbaines/projects/gatsby-eslint/_this_is_virtual_fs_path_/$virtual/sync-requires.js
  4:30  warning  Unexpected mix of '&&' and '||'  no-mixed-operators
  4:43  warning  Unexpected mix of '&&' and '||'  no-mixed-operators
```

<img width="1440" alt="Screenshot 2020-08-09 at 20 46 04" src="https://user-images.githubusercontent.com/24367010/89740593-030cde00-da82-11ea-8891-502039366eac.png">

### Steps to reproduce

```sh
npx gatsby new gatsby-eslint
cd gatsby-eslint
yarn add -D gatsby-plugin-eslint eslint-loader eslint eslint-plugin-react-app
touch .eslintrc
```

Add the following to `.eslintrc`:
```json
{
  "extends": ["plugin:react-app/recommended"]
}
```

Add the plugin to `gatsby-config.js`:
```diff
  plugins: [
+   `gatsby-plugin-eslint`,
    ...
```

```sh
yarn start
```

Minimal Reproduction: https://github.com/mrseanbaines/gatsby-eslint

### Expected result

No ESLint warnings for files that don't exist

### Actual result

ESLint warnings in file that doesn't exist

### Environment
```

  System:
    OS: macOS 10.15.6
    CPU: (8) x64 Intel(R) Core(TM) i5-8279U CPU @ 2.40GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.17.0 - ~/.nvm/versions/node/v12.17.0/bin/node
    Yarn: 1.22.4 - ~/.yarn/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v12.17.0/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 84.0.4147.105
    Firefox: 78.0.2
    Safari: 13.1.2
  npmPackages:
    gatsby: ^2.24.26 => 2.24.26 
    gatsby-image: ^2.4.14 => 2.4.14 
    gatsby-plugin-eslint: ^2.0.8 => 2.0.8 
    gatsby-plugin-manifest: ^2.4.21 => 2.4.21 
    gatsby-plugin-offline: ^3.2.21 => 3.2.21 
    gatsby-plugin-react-helmet: ^3.3.10 => 3.3.10 
    gatsby-plugin-sharp: ^2.6.24 => 2.6.24 
    gatsby-source-filesystem: ^2.3.23 => 2.3.23 
    gatsby-transformer-sharp: ^2.5.12 => 2.5.12 
```

I've got this too but this is my existing project since last year and just upgraded outdated packages to the latest version. Here is my log:

```sh
ROR #98123  WEBPACK

Generating development JavaScript bundle failed


/home/bluenex/gitRepo/_side-projects/neuromatch/frontend/_this_is_virtual_fs_path_/$virtual/sync-requires.js
   1:25   error    'react-hot-loader' should be listed in the project's dependencies. Run 'npm i -S react-hot-loader' to add it  import/no-extraneous-dependencies
   1:25   error    Strings must use singlequote                                                                                  quotes
   1:49   error    Missing semicolon                                                                                             semi
   4:23   error    Expected parentheses around arrow function argument                                                           arrow-parens
   4:30   warning  Unexpected mix of '&&' and '||'                                                                               no-mixed-operators
   4:43   warning  Unexpected mix of '&&' and '||'                                                                               no-mixed-operators
   4:47   error    Missing semicolon                                                                                             semi
   6:1    error    More than 1 blank line not allowed                                                                            no-multiple-empty-lines
   8:3    error    Strings must use singlequote                                                                                  quotes
   8:66   error    Do not import modules using an absolute path                                                                  import/no-absolute-path
   8:66   error    Strings must use singlequote                                                                                  quotes
   9:3    error    Strings must use singlequote                                                                                  quotes
   9:61   error    Do not import modules using an absolute path                                                                  import/no-absolute-path
   9:61   error    Strings must use singlequote                                                                                  quotes
  10:3    error    Strings must use singlequote                                                                           
```

I've got this too but this is my existing project since last year and just upgraded outdated packages to the latest version. Here is my log:

```sh
ERROR #98123  WEBPACK

Generating development JavaScript bundle failed


/home/.../_this_is_virtual_fs_path_/$virtual/sync-requires.js
   1:25   error    'react-hot-loader' should be listed in the project's dependencies. Run 'npm i -S react-hot-loader' to add it  import/no-extraneous-dependencies
   1:25   error    Strings must use singlequote                                                                                  quotes
   1:49   error    Missing semicolon                                                                                             semi
   4:23   error    Expected parentheses around arrow function argument                                                           arrow-parens
   4:30   warning  Unexpected mix of '&&' and '||'                                                                               no-mixed-operators
   4:43   warning  Unexpected mix of '&&' and '||'                                                                               no-mixed-operators
   4:47   error    Missing semicolon                                                                                             semi
   6:1    error    More than 1 blank line not allowed                                                                            no-multiple-empty-lines
   8:3    error    Strings must use singlequote                                                                                  quotes
   8:66   error    Do not import modules using an absolute path                                                                  import/no-absolute-path
   8:66   error    Strings must use singlequote                                                                                  quotes
   9:3    error    Strings must use singlequote                                                                                  quotes
   9:61   error    Do not import modules using an absolute path                                                                  import/no-absolute-path
   9:61   error    Strings must use singlequote                                                                                  quotes
  10:3    error    Strings must use singlequote                                                                           
```

--- a/packages/gatsby/src/bootstrap/requires-writer.ts
+++ b/packages/gatsby/src/bootstrap/requires-writer.ts
@@ -195,7 +195,7 @@ export const writeAll = async (state: IGatsbyState): Promise<boolean> => {
   let syncRequires = `${hotImport}
 
 // prefer default export if available
-const preferDefault = m => m && m.default || m
+const preferDefault = m => (m && m.default) || m
 \n\n`
   syncRequires += `exports.components = {\n${components
     .map(
@@ -209,7 +209,7 @@ const preferDefault = m => m && m.default || m
 
   // Create file with async requires of components/json files.
   let asyncRequires = `// prefer default export if available
-const preferDefault = m => m && m.default || m
+const preferDefault = m => (m && m.default) || m
 \n`
   asyncRequires += `exports.components = {\n${components
     .map((c: IGatsbyPageComponent): string => {

